omniledger/service/service.go

```diff
@@ -642,10 +659,19 @@ func (s *Service) verifySkipBlock(newID []byte, newSB *skipchain.SkipBlock) bool
 		return false
 	}
 
-	_, bodyI, err := network.Unmarshal(newSB.Payload, cothority.Suite)
-	body, ok := bodyI.(*DataBody)
-	if err != nil || !ok {
-		log.Error("couldn't unmarshal body", err, ok)
+	var body DataBody
+	err = protobuf.DecodeWithConstructors(newSB.Payload, &body, network.DefaultConstructors(cothority.Suite))
+	if err != nil {
+		log.Errorf("verifySkipblock: couldn't unmarshal body")
+		return false
+	}
+
+	rhash := sha256.New()
+	for _, x := range body.RefusedTransactions {
+		rhash.Write(x)
+	}
```

Comment:
Similar to my commend above, the follower isn't doing any real verification on the list of bad transactions, it's simply summing the hashes.